src/app/item-popup/ItemStat.tsx

```diff
@@ -41,50 +40,48 @@ export default function ItemStat({ stat, item }: { stat: DimStat; item?: DimItem
   const masterworkValue = item?.masterworkInfo?.stats?.[masterworkIndex]?.value || 0;
   const masterworkDisplayValue = (isMasterworkedStat && masterworkValue) || armor2MasterworkValue;
 
-  const moddedStatValue = item && getModdedStatValue(item, stat);
+  // const moddedStatValue = item && getTotalModEffects(item, stat);
+  const modEffects = item && _.sortBy(getModEffects(item, stat), ([n]) => -n);
+  const modEffectsTotal = modEffects ? _.sumBy(modEffects, ([n]) => n) : 0;
 
   const baseBar = item?.bucket.inArmor
-    ? Math.min(stat.base, stat.value)
-    : stat.value - masterworkValue;
-
-  const segments: [number, string?][] = [[baseBar]];
+    ? // if it's armor, the base bar length should be
+      // the shortest of base or resulting value, but not below 0
+      Math.max(Math.min(stat.base, stat.value), 0)
+    : // otherwise, for weapons, we just subtract masterwork and
+      // consider the "base" to include selected plugs
+      stat.value - masterworkValue;
+
+  //               amount,className,modName
+  const segments: [number, string?, string?][] = [[baseBar]];
```

Comment:
I think you can name tuple types now: `[amount: number, classname: string?, modName: string?]`

Reply:
thanks